src/main/drivers/accgyro/accgyro_spi_icm456xx.c

```diff
@@ -0,0 +1,641 @@
+           
+        }
+#endif
+        break;
+    }
+
+
+    case GYRO_EXTI_INT_DMA:
+    {
+        // If read was triggered in interrupt don't bother waiting. The worst that could happen is that we pick
+        // up an old value.
+
+        // This data was read from the gyro, which is the same SPI device as the acc
+        acc->ADCRaw[X] = (int16_t)((acc->gyro->dev.rxBuf[3] << 8) | acc->gyro->dev.rxBuf[2]);
+        acc->ADCRaw[Y] = (int16_t)((acc->gyro->dev.rxBuf[5] << 8) | acc->gyro->dev.rxBuf[4]);
+        acc->ADCRaw[Z] = (int16_t)((acc->gyro->dev.rxBuf[7] << 8) | acc->gyro->dev.rxBuf[6]);
+        break;
+    }
+
+    default:
+        break;
+    }
+
+    return true;    
+}
+ 
+bool icm456xxSpiAccDetect(accDev_t *acc)
+{
+    switch (acc->mpuDetectionResult.sensor) {
+    case ICM_45686_SPI:
+    case ICM_45605_SPI:
+        acc->initFn = icm456xxAccInit;
+        acc->readFn = icm456xxAccReadSPI;
+        break;
+    default:
+        return false;
+    }
+
+    return true;
+}
+
+bool icm456xxGyroReadSPI(gyroDev_t *gyro)
+{
+    switch (gyro->gyroModeSPI) {
+    case GYRO_EXTI_INIT:
+    {
+        // Initialise the tx buffer to all 0xff
+        memset(gyro->dev.txBuf, 0xff, ICM456XX_SPI_BUFFER_SIZE);
+
+        gyro->gyroDmaMaxDuration = 0; // INT gyroscope always calls that data is ready. We can read immediately
+#ifdef USE_DMA
+        if (spiUseDMA(&gyro->dev)) {
+            gyro->dev.callbackArg = (uint32_t)gyro;
+            gyro->dev.txBuf[0] = ICM456XX_GYRO_DATA_X1_UI | 0x80;
+            gyro->segments[0].len = ICM456XX_SPI_BUFFER_SIZE;
+            gyro->segments[0].callback = mpuIntCallback;
+            gyro->segments[0].u.buffers.txData = gyro->dev.txBuf;
+            gyro->segments[0].u.buffers.rxData = &gyro->dev.rxBuf[1];
+            gyro->segments[0].negateCS = true;
+            gyro->gyroModeSPI = GYRO_EXTI_INT_DMA;
+        } else
+#endif
+        {
+            // Interrupts are present, but no DMA. Non-DMA read
+            uint8_t raw[ICM456XX_DATA_LENGTH];
+            const bool ack = spiReadRegMskBufRB(&gyro->dev, ICM456XX_GYRO_DATA_X1_UI, raw, ICM456XX_DATA_LENGTH);
+            if (!ack) {
+                return false;
+            }
+
+            gyro->gyroADCRaw[X] = (int16_t)((raw[1] << 8) | raw[0]);
+            gyro->gyroADCRaw[Y] = (int16_t)((raw[3] << 8) | raw[2]);
+            gyro->gyroADCRaw[Z] = (int16_t)((raw[5] << 8) | raw[4]);
+            gyro->gyroModeSPI = GYRO_EXTI_INT;
+        }
+
+        break;
+    }
+
+    case GYRO_EXTI_NO_INT:
+    {
+        gyro->dev.txBuf[0] = ICM456XX_GYRO_DATA_X1_UI | 0x80;
+
+        busSegment_t segments[] = {
+                {.u.buffers = {NULL, NULL}, ICM456XX_SPI_BUFFER_SIZE, true, NULL},
+                {.u.link = {NULL, NULL}, 0, true, NULL},
+        };
+        memset(&gyro->dev.txBuf[1], 0xFF, 6);
+        segments[0].u.buffers.txData = gyro->dev.txBuf;
+        segments[0].u.buffers.rxData = &gyro->dev.rxBuf[1];
```

Comment:
No need for `[1]` here (except to align gyro data to even address) - simply use rxBuff, than skip (only) first byte
```suggestion
        segments[0].u.buffers.rxData = gyro->dev.rxBuf;
```

This is repeated in other rxBuf uses

Reply:
You are right! Copy-paste from another gyroscope didn't think about buffers at first, already figured out SPI DMA API when debugging SPI NAND flash